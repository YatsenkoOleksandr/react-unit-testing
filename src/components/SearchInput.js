export const SearchInput = ({ value, onChange, children }) => {
  return (
    <div>
      <label htmlFor="search">{children}</label>
      <input
        id="search"
        type="text"
        value={value}
        onChange={(event) => onChange(event.target.value)} />
    </div>
  );
};
