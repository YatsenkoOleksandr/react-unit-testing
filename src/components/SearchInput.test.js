import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { SearchInput } from './SearchInput';

describe('SearchInput', () => {
  describe('Enzyme', () => {
    test('should render label with text', () => {
      // Arrange
      const labelText = 'Test';

      // Act
      const wrapper = shallow(<SearchInput value=''>{labelText}</SearchInput>);

      // Assert
      expect(wrapper.find(`label[htmlFor="search"]`).text()).toEqual(labelText);
    });

    test('should render search input with value', () => {
      // Arrange
      const value = 'Test'

      // Act
      const wrapper = shallow(<SearchInput value={value} />);

      // Assert
      expect(wrapper.find(`input[type="text"]`).props().value).toEqual(value);
    });

    test('should call onChange when text changed in search input', () => {
      // Arrange
      const value = ''
      const changedValue = 'Changed';
      const onChange = jest.fn();

      // Act
      const wrapper = shallow(<SearchInput value={value} onChange={onChange} />);
      wrapper.find(`input[type="text"]`).simulate('change', { target: { value: changedValue } });

      // Assert
      expect(onChange).toBeCalledWith(changedValue);
    });
  });

  describe('Snapshot', () => {    
    test('should match snapshot and have label with text', () => {
      // Arrange
      const labelText = 'Test Label';
      const value = '';
      const onChange = jest.fn();

      // Act
      const tree = renderer
        .create(<SearchInput value={value} onChange={onChange}>{labelText}</SearchInput>)
        .toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });

    test('should match snapshot and have search input with value', () => {
      // Arrange
      const labelText = '';
      const value = 'Test Input';
      const onChange = jest.fn();

      // Act
      const tree = renderer
        .create(<SearchInput value={value} onChange={onChange}>{labelText}</SearchInput>)
        .toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });

  describe('React Testing Library', () => {
    test('should render label with text', () => {
      // Arrange
      const labelText = 'Test';

      // Act
      render(<SearchInput value=''>{labelText}</SearchInput>);

      // Assert
      expect(screen.getByLabelText(labelText)).toBeInTheDocument();
    });

    test('should render search input with value', () => {
      // Arrange
      const value = 'Test'

      // Act
      render(<SearchInput value={value} />);

      // Assert
      expect(screen.getByDisplayValue(value)).toBeInTheDocument();
    });

    // test('should call onChange when text changed in search input', () => {
    //   // Arrange
    //   const value = ''
    //   const changedValue = 'Changed';
    //   const onChange = jest.fn();

    //   // Act
    //   render(<SearchInput value={value} onChange={onChange} />);
    //   userEvent.type(screen.getByRole('textbox'), changedValue);

    //   // Assert
    //   expect(onChange).toBeCalledWith(changedValue);
    // });
  });
});