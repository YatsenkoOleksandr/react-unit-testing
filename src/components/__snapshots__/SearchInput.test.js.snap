// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Snapshot should match snapshot and have label with text 1`] = `
<div>
  <label
    htmlFor="search"
  >
    Test Label
  </label>
  <input
    id="search"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`SearchInput Snapshot should match snapshot and have search input with value 1`] = `
<div>
  <label
    htmlFor="search"
  >
    
  </label>
  <input
    id="search"
    onChange={[Function]}
    type="text"
    value="Test Input"
  />
</div>
`;
